src/libertem/job/masks.py

```diff
@@ -83,8 +89,37 @@ def _compute_masks(self):
         -------
         a list of masks as they were created by the factories
         """
-        return [f().astype(self.dtype)
-                for f in self.mask_factories]
+        # Make sure all the masks are either sparse or dense
+        # If the use_sparse property is set to Ture or False,
+        # it takes precedence.
+        # If it is None, use sparse only if all masks are sparse
+        # and set the use_sparse property accordingly
+        def to_dense(a):
+            if sp.issparse(a):
+                return a.toarray()
+            else:
+                return a
+
+        def to_sparse(a):
+            if sp.issparse(a):
+                return a
+            else:
+                return sp.csr_matrix(a)
+        raw_masks = [f().astype(self.dtype)
+            for f in self.mask_factories]
```

Comment:
Minor nitpick: I like to have multi-line list comprehensions aligned, maybe something like this:

```suggestion
        raw_masks = [
            f().astype(self.dtype)
            for f in self.mask_factories
        ]
```

(hope this suggestion stuff actually works, I think it doesn't get that I'm replacing two lines here...)

Reply:
A new version is coming with next commit, I hope it is what you meant?